src/extensions/data_array.js

```diff
@@ -67,24 +96,101 @@ export default () => {
       });
     };
 
+    const createInstances = observationObjects => {
+      return new Promise(resolve => {
+        db().then(models => {
+          Promise.all(observationObjects.map(observationObj => {
+            const newReq = Object.create(req);
+            newReq.body = observationObj.body;
+            return models.createInstance(observations, newReq, exclude).then(
+              instance => {
+              observationObj.instanceId = instance.id;
+              return Promise.resolve();
+            }).catch(() => {
+              return Promise.resolve();
+            });
+          })).then(results => {
+            return resolve(results);
+          });
+        }).catch(() => {
+          return ERR.ApiError(res, 500,
+            ERR.ERRNO_INTERNAL_ERROR, ERR.INTERNAL_ERROR);
+        });
+      });
+    };
+
+    const createObservations = body => {
+      let observationObjects = [];
+      const resultIndex = body.components.indexOf(result);
+      const featureOfInterestIndex =
+        body.components.indexOf(featureOfInterest + '/id');
```

Comment:
That constant is defined as ```'FeatureOfInterestId'```. I couldn't think of a proper name for ```'FeatureOfInterestId/id'``` so I decided to essentially scope ```'result'```, ```'phenomenonTime'```, and ```'FeatureOfInterestId/id'``` by creating a ```components``` object.